clippy_lints/src/methods/sliced_string_as_bytes.rs

```diff
@@ -0,0 +1,30 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::source::snippet_with_applicability;
+use clippy_utils::ty::is_type_lang_item;
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind, LangItem, is_range_literal};
+use rustc_lint::LateContext;
+
+use super::SLICED_STRING_AS_BYTES;
+
+pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, recv: &Expr<'_>) {
+    if let ExprKind::Index(indexed, index, _) = recv.kind
+        && is_range_literal(index)
+    {
+        let ty = cx.typeck_results().expr_ty(indexed).peel_refs();
```

Comment:
Note that this `let` can be part of the preceding `if … && … && …` chain, as irrefutable `let` patterns can be used there as well. This would remove one level of indentation from lines 16 to 27.

Reply:
Thanks, addressed!